### ISSUE TITLE
Contract Board Bug

### ISSUE THREAD

**Issue by Chriskropp** [StrangeLoopGames/EcoIssues#8780]
If you create a contract and no matter what Invetory connects you can do the job without the need to submit even 1 item.

It is easy to pick up and store the required item. No matter where it is, no matter if it's in a warehouse or a vehicle you take it on Hand and put it back, and it got counted for the Contract.

So actually the Contract put item in Container cant be used...

Best Regards

**Comment by johnkslg** [comment]
Not sure what the problem is, I tried a contract with Put Items In Container, and marked finished before I did anything, but it failed me:

![image](https://user-images.githubusercontent.com/3536496/43429550-cffe340c-9418-11e8-8cfe-6ad431cb153d.png)

Reopen if you have details how to repro this.

**Comment by TheeJudge** [comment]
You have container A, You need 10 stone in container A so you make a contract using "Put items in container". I accept that contract, go to my stockpile and grab 10 stone and put in my truck to take it to there container. The contract is done and I get paid.

The problem is that the contractor wants the items in container A but all i had to do was put those required items into my truck and I get paid. The person who made the contract will never get the items they wanted but will lose their money.

If there were multiple of these then it could be exploited badly to take a lot of money from someone without ever getting what they wanted.

**Comment by Zimgaha** [comment]
I see this issue has been closed, however i am having the same issue.

I set up a contract as follows;

![20181106200433_1](https://user-images.githubusercontent.com/44818253/48090612-4b3d0380-e1ff-11e8-9c27-f7e929f5a129.jpg)

Now when the player accepts the contract, and starts to do the job, soon as they put the stone in their cart it counts towards the contract amount even tho they have not deposited the item in the destination. We have done some testing and it also seems to count if they add the stone to a random stockpile (not the one assigned as deposit) meaning in reality they could add stone to a cart, remove it, add it, remove it so on and so on until the contact marks as complete.

We have tried server restarts, logins, numerous contacts, different locations, different ownership permissions for the land, we just cant seem to get round it  .
